include/nn_dot_product_matrix.h

```diff
@@ -0,0 +1,28 @@
+#ifndef NN_DOT_PRODUCT_MATRIX_H
+#define NN_DOT_PRODUCT_MATRIX_H
+
+#include "nn_error.h"
+#include <stddef.h>
+
+// MATRIX_MAX_ROWS defines the maximum number of rows in a matrix.
+#ifndef MATRIX_MAX_ROWS
+#define MATRIX_MAX_ROWS 3
+#endif
+
+// MATRIX_MAX_COLS defines the maximum number of columns in a matrix.
+#ifndef MATRIX_MAX_COLS
+#define MATRIX_MAX_COLS 3
+#endif
+
+// NNDotProductMatrixFunction represents a function that calculates
+// the dot product of two matrices.
+typedef void (*NNDotProductMatrixFunction)(float result[MATRIX_MAX_ROWS][MATRIX_MAX_COLS], const float a[MATRIX_MAX_ROWS][MATRIX_MAX_COLS], const float b[MATRIX_MAX_ROWS][MATRIX_MAX_COLS]);
```

Comment:
Need to rename it to `NNDotProdMatrixFunc` be consistent with https://github.com/devfacet/nntbn/blob/4c635a24c3851988130fe81c6a6e42bf853df34e/include/nn_dot_product.h#L12

Also the function signature doesn't seem right. It should be something like:

```
void nn_dot_prod_matrix(const float a[NN_DOT_PROD_MATRIX_MAX_ROW_SIZE][NN_DOT_PROD_MATRIX_MAX_COL_SIZE], const float b[NN_DOT_PROD_MATRIX_MAX_ROW_SIZE][NN_DOT_PROD_MATRIX_MAX_COL_SIZE], float output[NN_DOT_PROD_MATRIX_MAX_ROW_SIZE][NN_DOT_PROD_MATRIX_MAX_COL_SIZE], size_t row_size_a, size_t col_size_a, size_t row_size_b, size_t col_size_b)
```

It should be drop in replacement to https://github.com/devfacet/nntbn/blob/4c635a24c3851988130fe81c6a6e42bf853df34e/src/nn_layer.c#L123-L128 and then we just add biases.

I would check numpy.dot function which supports both vector (we already have it) and matrix.